x-pack/plugins/reporting/server/lib/store/report.ts

```diff
@@ -7,57 +7,98 @@
 // @ts-ignore no module definition
 import Puid from 'puid';
 
-interface Payload {
-  id?: string;
-  index: string;
+/*
+ * The document created by Reporting to store in the .reporting index
+ */
+interface ReportingDocument {
+  _id: string;
+  _index: string;
+  _seq_no: unknown;
+  _primary_term: unknown;
   jobtype: string;
-  created_by: string | boolean;
+  created_by: string | null;
   payload: unknown;
+  meta: unknown;
   browser_type: string;
-  priority: number;
   max_attempts: number;
   timeout: number;
+
+  status: string;
+  attempts: number;
+  output?: unknown;
+  started_at?: string;
+  completed_at?: string;
+  created_at?: string;
+  priority?: number;
+  process_expiration?: string;
+}
+
+/*
+ * The document created by Reporting to store as task parameters for Task
+ * Manager to reference the report in .reporting
+ */
+export interface ReportTaskJSON {
+  id: string;
+  index: string;
+  jobtype: string;
 }
 
 const puid = new Puid();
 
 export class Report {
   public readonly jobtype: string;
-  public readonly created_by: string | boolean;
+  public readonly created_at: string;
+  public readonly created_by: string | null;
   public readonly payload: unknown;
+  public readonly meta: unknown;
   public readonly browser_type: string;
-  public readonly id: string;
 
-  public readonly priority: number;
   // queue stuff, to be removed with Task Manager integration
   public readonly max_attempts: number;
   public readonly timeout: number;
 
-  public _index: string;
+  public _index?: string;
   public _id?: string; // set by ES
```

Comment:
Previously, there was an `id` and `_id`. I changed the structure to be less of an abstraction on top of the data in ES. The main thing now is the fields are all one level, but in ES most of the fields are nested in `_source`